### ISSUE TITLE
Switch from Karma to WebdriverIO

### ISSUE THREAD

**Issue by christian-bromann** [angular/angular-cli#25547]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

No

### Description

It seems that Karma is not much maintained anymore as browser test framework and I wonder if you team would be interested switching to WebdriverIO.

I've been working on supporting unit and component based browser tests in WebdriverIO within the recent month and would love to battle test it against mature projects like Angular.

I am happy to do everything need to switch the harness. I feel like the setup would become much easier and it would be also very easy to run tests across different browser if this is desired.

### Please provide a link to a minimal reproduction of the bug

m/a

### Please provide the exception or error you saw

```true
n/a
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
n/a
```

**Comment by dgp1130** [comment]
@christian-bromann, thanks for bringing this up. I don't think I was aware that WebDriverIO supported component-level tests in a real browser.

We recently [updated Angular's unit testing strategy](https://blog.angular.io/moving-angular-cli-to-jest-and-web-test-runner-ef85ef69ceca) based on Karma's deprecation and are somewhat committed to migrating to Web Test Runner and Jest as alternatives. WTR in particular is intended to fill Karma's specific niche with respect to in-browser unit testing. Unfortunately, I don't expect we'd have the bandwidth to maintain a 3rd officially supported unit testing pipeline within Angular if that's what you're looking for here.

We definitely love to see the broader web ecosystem adopting support in Angular and we're happy to help enable and support the community the best we can. It looks like WebDriverIO already supports a wide number of frameworks, are there any particular challenges or issues which make Angular support impractical compared to other frameworks?

**Comment by christian-bromann** [comment]
Thanks for responding!

Web Test Runner is a great project and I looked into it when I started to implement component-level testing in WebdriverIO. I think it provides more or less the same capabilities with differences in details (Vite vs Sinon style mocking).

> WTR in particular is intended to fill Karma's specific niche with respect to in-browser unit testing.

WebdriverIO does so too 😉

> are there any particular challenges or issues which make Angular support impractical compared to other frameworks?

There is an open issue for it to add Angular presets to WebdriverIO component testing, so that will land soon.

Feel free to close this.
